docs/build/wasm/interact/interact.md

```diff
@@ -1,5 +1,139 @@
+
+### Installation
+
+```bash
+yarn add @polkadot/api@9.13.6 @astar-network/astar-api@0.1.14
+```
+
+Example: Here is a dapp example that uses Astar.js to interact with WASM smart contract. You can find the source code [dApp](https://github.com/astarNetwork/wasm-lottery)
+This is a simple lottery dapp that allows users to enter and draw the lottery. The lottery is implemented as a WASM smart contract.
+
+### Usage
+
+#### Connecting to API
+
+```js
+import { ApiPromise } from '@polkadot/api';
+import { WsProvider } from '@polkadot/rpc-provider';
+import { options } from '@astar-network/astar-api';
+import { sendTransaction } from '@astar-network/astar-sdk-core';
+
+async function main() {
+    const provider = new WsProvider('ws://localhost:9944');
+    // OR
+    // const provider = new WsProvider('wss://shiden.api.onfinality.io/public-ws');
+    const api = new ApiPromise(options({ provider }));
+    await api.isReady;
+
+    // Use the api
+    // For example:
+    console.log((await api.rpc.system.properties()).toHuman());
+
+    process.exit(0);
+}
+
+main()
+```
+
+#### Initialise Contract Class
+
+```js
+import { Abi, ContractPromise } from '@polkadot/api-contract'
+
+// After compiling the contract a ABI json is created in the artifacts. Import the ABI:
+// If you didn't use swanky the name will be metadata.json
+import ABI from './artifacts/lottery.json'
+
+const abi = new Abi(ABI, api.registry.getChainProperties())
+
+// Initialise the contract class
+const contract = new ContractPromise(api, abi, address) // address is the deployed contract address
+```
+
+#### Query Contract Messages
+
+```js
+
+// Get the gas WeightV2 using api.consts.system.blockWeights['maxBlock']
+const gasLimit = api.registry.createType(
+    'WeightV2',
+    api.consts.system.blockWeights['maxBlock']
+  )
+
+// Query the contract message
+const { gasRequired, result, output } = await contract.query.pot(
+    account.address,
+    {
+      gasLimit,
+    }
+  )
+```
+
+#### Send Contract Transaction
+
+Sending contract transaction normally is a 2 step process. First is to query the transaction and check for errors. We have a helper function to do this for you. The second step is to send the transaction. The helper function will return the transaction object which you can use to sign and send the transaction.
+The parameters for the helper function are:
+
+- api: The api instance
+- contract: The contract instance
+- message: The message to send
+- sender: The sender address
+- value: The value to send with the transaction
+- ...params: The parameters for the message
+
+```js
+
+import { sendTransaction } from '@astar-network/astar-sdk-core';
+
+try {
+  const result = await sendTransaction(api, contract, 'enter', account.address, new BN('1000000000000000000'))
```

Comment:
* What if contract method takes more parameters?
* Can you rewrite this line to have parameters in separate lines and add comment on each param

```
await sendTransaction(
param1, // comment param 1
param2, // comment param 2
)
```